scylla.go

```diff
@@ -484,53 +488,128 @@ func (p *scyllaConnPicker) GetCustomDialer() Dialer {
 		shardID := (p.lastAttemptedShard + i) % p.nrShards
 		if p.conns == nil || p.conns[shardID] == nil {
 			p.lastAttemptedShard = shardID
-			return &scyllaOneShardDialer{
-				shardAwareAddress: p.shardAwareAddress,
-				shardID:           shardID,
-				nrShards:          p.nrShards,
-				dialer:            p.dialer,
-			}
+			return shardID, p.nrShards
 		}
 	}
 
 	// We did not find an unallocated shard
 	// We will dial the non-shard-aware port
-	return nil
+	return 0, 0
+}
+
+// ShardDialer is like HostDialer but is shard-aware.
+// If the driver wants to connect to a specific shard, it will call DialShard,
+// otherwise it will call DialHost.
+type ShardDialer interface {
+	HostDialer
+
+	// DialShard establishes a connection to the specified shard ID out of nrShards.
+	// The returned connection must be directly usable for CQL protocol,
+	// specifically DialShard is responsible also for setting up the TLS session if needed.
+	DialShard(ctx context.Context, host *HostInfo, shardID, nrShards int) (DialedHost, error)
```

Comment:
```suggestion
	DialShard(ctx context.Context, host *HostInfo, shardID, nrShards int) (*DialedHost, error)
```
It's easier to work with, and there's no need for copying.